From dfed18c3e7e662d77ccf1d314f4399f1fdaa48f8 Mon Sep 17 00:00:00 2001
From: Shaun Walbridge <shaun.walbridge@gmail.com>
Date: Tue, 25 Feb 2025 22:15:42 -0500
Subject: [PATCH 2/4] Use upstream `cloudpickle`

There are no modifications made to the `cloudpickle` package, rather
than vendoring it, allow pip resolution to find the updated package.
Upstream has a number of improvements, including explicit Python 3.13
support in its tests.
---
 setup.cfg            | 4 +---
 srsly/_pickle_api.py | 2 +-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index fe5f34a..dc94fa8 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -32,6 +32,7 @@ setup_requires =
     cython>=0.29.1
 install_requires =
     catalogue>=2.0.3,<2.1.0
+    cloudpickle>=2.2.0
 
 [options.entry_points]
 # If spaCy is installed in the same environment as srsly, it will automatically
@@ -59,6 +60,3 @@ exclude =
 
 [mypy]
 ignore_missing_imports = True
-
-[mypy-srsly.cloudpickle.*]
-ignore_errors=True
diff --git a/srsly/_pickle_api.py b/srsly/_pickle_api.py
index 0e894d9..d8f6d80 100644
--- a/srsly/_pickle_api.py
+++ b/srsly/_pickle_api.py
@@ -1,6 +1,6 @@
 from typing import Optional
 
-from . import cloudpickle
+import cloudpickle
 from .util import JSONInput, JSONOutput
 
 
